### ISSUE TITLE
Issue on base_lr of cosine LR of LRScheduler

### ISSUE THREAD

**Issue by zx33** [dmlc/gluon-cv#1350]
```
Gluoncv 0.6.0
MXNet 1.5.0
Windows 10 18363
Python 3.6.6
Cuda 9.0
```

While using the LRScheduler setting cosine learning rate, `base_lr` param is set to 0.1, but the actual value of the starting LR is 0.01.

Sample code
```
from __future__ import division

import argparse, time, logging, random, math, os
import numpy as np
import mxnet as mx

from mxnet import gluon, nd
from mxnet import autograd as ag
from mxnet.gluon import nn
from mxnet.gluon.data.vision import transforms

from gluoncv.utils import makedirs, TrainingHistory, LRSequential, LRScheduler
from gluoncv.data import transforms as gcv_transforms

net = nn.HybridSequential()
net.add(nn.Conv2D(16, 3))
net.hybridize()
net.initialize(mx.init.MSRAPrelu(), ctx=mx.gpu(0))

batch_size = 128
epochs = 50
num_batches = 50000 // batch_size
optimizer = 'nag'
lr_scheduler = LRScheduler('cosine', base_lr=0.1, target_lr=0, nepochs=epochs, iters_per_epoch=num_batches, step_factor=1)
optimizer_params = {'wd': 0.0001, 'momentum': 0.9, 'lr_scheduler': lr_scheduler}
trainer = gluon.Trainer(net.collect_params(), optimizer, optimizer_params)

print('debug:', trainer.learning_rate)
```

Debug outputs:

```
debug: 0.01
```

However, while using LRSequential with the same LRScheduler setting, this issue does not happen.

```
# lr_scheduler = LRScheduler('cosine', base_lr=0.1, target_lr=0, nepochs=epochs, iters_per_epoch=num_batches, step_factor=1)
lr_scheduler = LRSequential([
    LRScheduler('cosine', base_lr=0.1, target_lr=0, nepochs=epochs, iters_per_epoch=num_batches)
])
```
